File: 1_GuidedExercise/idea_03.rb
class Idea03
  # Task: Currently we recalculate the average number every iteration. Extract it into a variable.
  def introduce_a_variable(number_list)
    big_numbers = []

    number_list.each do |number|
      if number > number_list.reduce(0, :+) / number_list.length
        big_numbers.push(number)
      end
    end

    big_numbers
  end
end

File: js/1_GuidedExercise/src/idea_03.js
// Task: Currently we recalculate the average number every iteration. Extract it into a variable.
const introduceAVariable = (numberList) => {
    if (numberList.length === 0) {
        return []
    }

    const bigNumbers = []

    numberList.forEach((number) => {
        if (number > numberList.reduce((accumulator, current) => accumulator + current, 0) / numberList.length) {
            bigNumbers.push(number)
        }
    })

    return bigNumbers
}

module.exports = {introduceAVariable}
